src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -263,30 +260,31 @@ void Runnable::begin(Callback<Error> cb) {
                         cb(MissingRequiredInputFileError());
                         return;
                     }
-                    input_generator.reset(new std::ifstream(input_filepath));
-                    if (!input_generator->good()) {
+                    std::ifstream input_generator{input_filepath};
+                    if (!input_generator.good()) {
                         logger->warn("cannot read input file");
                         cb(CannotOpenInputFileError());
                         return;
                     }
 
                     // Count the number of entries
                     std::string next_input;
-                    while ((std::getline(*input_generator, next_input))) {
+                    while ((std::getline(input_generator, next_input))) {
                         num_entries += 1;
+                        inputs.push_back(next_input);
                     }
-                    if (!input_generator->eof()) {
+                    if (!input_generator.eof()) {
                         logger->warn("cannot read input file");
                         cb(FileIoError());
                         return;
                     }
-                    // See http://stackoverflow.com/questions/5750485
-                    //  and http://stackoverflow.com/questions/28331017
-                    input_generator->clear();
-                    input_generator->seekg(0);
-
+                    // See http://en.cppreference.com/w/cpp/algorithm/shuffle
+                    std::random_device rd;
+                    std::mt19937 g(rd());
+                    std::shuffle(inputs.begin(), inputs.end(), g);
```

Comment:
Can we make the shuffling of inputs something that is enabled optionally based on a flag? (Maybe the flag could be on by default, but it should be possible to change the behaviour).